scripts/partition_manager.py

```diff
@@ -565,22 +614,42 @@ def parse_args():
     parser.add_argument("--input-files", required=True, type=str, nargs="+",
                         help="List of paths to input yaml files. ")
 
-    parser.add_argument("--flash-size", required=True, type=int,
-                        help="Flash size of chip in kB.")
+    parser.add_argument("--output-partitions", required=True, type=str,
+                        help="Path to output partition configuration file.")
+
+    parser.add_argument("--output-regions", required=True, type=str,
+                        help="Path to output regions configuration file.")
 
-    parser.add_argument("--flash-start", type=lambda x: int(x, 0), default=0,
-                        help="Start address of flash.")
+    parser.add_argument("--static-config", required=False, type=argparse.FileType(mode='r'),
+                        help="Path static configuration.")
 
-    parser.add_argument("--output", required=True, type=str,
-                        help="Path to output file.")
+    parser.add_argument("--regions", required=False, type=str, nargs='*',
+                        help="Space separated list of regions. For each region specified here, one must specify"
+                             "--{region_name}-base-addr and --{region_name}-size. If the region is associated"
+                             "with a driver, the device name must be given in --{region_name}-device (e.g. an "
+                             "external flash driver. For regions with placement strategy 'complex' it is possible"
+                             "to specify the --{region_name}-dynamic-partition to set the name of the dynamic partition"
+                             "which occupy all non-used area.")
 
-    parser.add_argument("-d", "--dynamic-partition", required=False, type=str,
-                        help="Name of dynamic partition")
+    known, ranges_arguments = parser.parse_known_args()
```

Comment:
Do you have a proposal for a different name? I struggle to find anything better. What I could do is to create a new namespace instead of simply smashing the new arguments into known before returning it.

Reply:
I split it into two, `known` and `region_configuration`.

Reply:
Changed to main_args and region_args.